rpcserver.go

```diff
@@ -4324,24 +4324,26 @@ func (r *rpcServer) ListPayments(ctx context.Context,
 			continue
 		}
 
-		// Fetch the payment's route, which will be empty if an attempt
-		// has not been made.
-		var route route.Route
-		if payment.Attempt != nil {
-			route = payment.Attempt.Route
+		// Fetch the payment's route and preimage. If no HTLC was
+		// successful, an empty route and preimage will be used.
+		var (
+			route    route.Route
+			preimage lntypes.Preimage
+		)
+		for _, htlc := range payment.HTLCs {
+			if htlc.Settle != nil {
+				route = htlc.Route
```

Comment:
Didn't we previously also expose the path for failed payments? Or was that only in `routerrpc`?

Reply:
fixed, we now show a route for failed+successful